pkg/controller/eventinstance/eventinstance.go

```diff
@@ -0,0 +1,91 @@
+package eventinstance
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultTTL = 24 * time.Hour
+
+func GCExpired() router.HandlerFunc {
+	// last stores pre-parsed TTLs from the configuration.
+	var last struct {
+		sync.Mutex
+		raw string
+		ttl time.Duration
+	}
+
+	return handler{
+		ttl: func(ctx context.Context, getter kclient.Reader) (time.Duration, error) {
+			cfg, err := config.Get(ctx, getter)
+			if err != nil {
+				return 0, err
+			}
+
+			cfgTTL := cfg.EventTTL
+			if cfgTTL == nil || *cfgTTL == "" {
+				return defaultTTL, nil
+			}
+
+			last.Lock()
+			defer last.Unlock()
+
+			if last.raw != *cfgTTL {
+				// This is a new TTL, parse and memoize
+				ttl, err := time.ParseDuration(*cfgTTL)
+				if err != nil {
+					return 0, err
+				}
+
+				last.raw, last.ttl = *cfgTTL, ttl
+			}
+
+			return last.ttl, nil
+		},
+	}.gcExpired
+}
+
+// ttlFunc is a function that returns the TTL to use for event expiration.
+type ttlFunc func(context.Context, kclient.Reader) (time.Duration, error)
+
+type handler struct {
+	ttl ttlFunc
```

Comment:
nit: calling this ttl confused me bc you also `ttl` for the duration variable. is there anything else you could name it to disambiguate it for clarity

Reply:
sure, will update.

Reply:
done